Fix example dependency list

tcharding opened the pull request:
Example relies on `rand-std` not plain `rand` dependency.

I do not understand why the following command passes without this patch
applied

```
cargo test --no-default-features --features=std,rand,bitcoin_hashes
```

But if we put the same code in a standalone binary it fails as expected?

Since the running of this test is _unusual_ and it is primarily meant as
an entry point example to the library, remove the mention of "alloc"
feature and just depend upon "std".

Fixes: #395

apoelstra reviewed (approved):
ACK 3c9dd2fb326b4ffe9680885d1de79904499a4043

If I understand the issue correctly, this example would've have worked with `alloc` anyway, since it depends on `std` implicitly through `rand-std`.

tcharding commented:
Its the "rand" vs "rand-std" thing that this PR really fixes.

I may have confused things by removing "alloc" at the same time, my bad. I dropped the "alloc" because I'm wondering if all the recent feature gates I changed were incorrect, it seems confusing to _not_ have "std" enabled but have "rand-std" enabled, even if technically this all works. Why would anyone ever do this. But changing them all _again_ seems like code churn, not sure what to do?
